jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/StdErrAppender.java

```diff
@@ -0,0 +1,331 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.logging;
+
+import java.io.PrintStream;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Objects;
+import java.util.TimeZone;
+
+import org.slf4j.event.Level;
+import org.slf4j.helpers.FormattingTuple;
+import org.slf4j.helpers.MessageFormatter;
+import org.slf4j.helpers.NormalizedParameters;
+
+public class StdErrAppender implements JettyAppender
+{
+    /**
+     * Configuration keys specific to the StdErrAppender
+     */
+    public static final String NAME_CONDENSE_KEY = "org.eclipse.jetty.logging.appender.NAME_CONDENSE";
+    public static final String THREAD_PADDING_KEY = "org.eclipse.jetty.logging.appender.THREAD_PADDING";
+    public static final String MESSAGE_ESCAPE_KEY = "org.eclipse.jetty.logging.appender.MESSAGE_ESCAPE";
+    public static final String STRICT_SLF4J_FORMAT_KEY = "org.eclipse.jetty.logging.appender.STRICT_SLF4J_SYNTAX";
+    public static final String ZONEID_KEY = "org.eclipse.jetty.logging.appender.ZONE_ID";
+
+    private static final Object[] EMPTY_ARGS = new Object[0];
+    private final DateTimeFormatter timestampFormatter;
```

Comment:
Why are we not using `DateCache`?    I know it is not as flexible as a modern formatter, but it is fast for the simple data stamp that this log needs.  If users want anything fancier, they should replace our logging.

Reply:
Our DateCache exists in jetty-util (which we cannot depend on), and it uses the standard Java DateTimeFormatter in exactly the same way like this code does (when formatting with milliseconds like this Appender does).
The DateCache only works on caching results that are on the same second and do not contain a millisecond formatting string.

Reply:
StdErrLog uses DateCache and separately handles the ms.  So why can't we depend on jetty-util?      I know it is a very old class.... but when it was introduced it was VERY beneficial.  If we are doing full logging, we can avoid thousands of time formats per second - well worth it!

Reply:
Yes, and I tested StdErrLog's DateCache usage, and found it has a 100% cache-miss on uses with DateCache simply because it formats with milliseconds.

Reply:
Opened #4605 to review DateCache.
The results of that can determine what we do here.
Use it more aggressively everywhere (including this jetty-slf4j-impl), or something else.

Reply:
Then let's not remove it and have to review to have it replace.  Keep it and review to see if we remove it!

Reply:
I cannot add the dependency on jetty-util here as jetty-util uses logging it's a loop in both the dependencies on maven and the load order at jvm runtime.
I would rather just rip out the DateTimeFormatter entirely and just use raw inline formatting against the Java 8+ date/time APIs.

Reply:
StdErrLog uses it, so this can use it.    Avoiding calling any formatter thousands of times per second is very valuble.

Reply:
I just ran StdErrLogTest and it had 1 MISS and every other log in the test was a date cache hit!

Reply:
Note the solution to the dependency issue is to make this implementation part of org.eclipse.jetty.util.log in jetty-util

Reply:
I did this differently:

* Brought in DateCache into jetty-sfl4j-impl
* Stripped out everything that jetty-slf4j-impl wouldn't use (down to 1 public method)
* Renamed it to Timestamp
* Have millisecond formatting done within Timestamp on top of the cached Tick.formattedString (if a cache hit or cache miss on the second cache, just like DateCache does)

Reply:
Oh but we can't do that... because then we can't replace the jar.... hmmmmm

Reply:
This is the only method in DateCache that a logger would use.

https://github.com/eclipse/jetty.project/blob/7ce14a419fd1c7cec42d808ac3e6fd862ce517b4/jetty-util/src/main/java/org/eclipse/jetty/util/DateCache.java#L203-L222

Incidentally, in the process, I discovered a bug in DateCache, filed in #4606